runner/runner.go

```diff
@@ -927,6 +928,25 @@ retry:
 		fullURL = parsedURL.String()
 	}
 
+	var asnResponse interface{ String() string }
+	if r.options.Asn {
+		addr, err := net.LookupIP(domain)
```

Comment:
We can skip this lookup by moving the logic later and reusing the `ip` variable value at https://github.com/projectdiscovery/httpx/blob/1ca3fb668c64f0b3a0016b5850647ed44c9f3bb1/runner/runner.go#L1130